### DISCUSSION TITLE
How to execute custom tool on macOS

### DISCUSSION THREAD

**Discussion by Dav-d** [zim-desktop-wiki/zim-desktop-wiki#2683] [Q&A]
Hi,

I use Zim for a long time on Linux. And recently I 've migrated to macOS.
I use zim with some custom tools as  shell script to scan and import my documents. It worked like a charm on Linux. 

But I can't manage to work it on macOS. 

I'm a newbe on macOS and I'm not aware with all the tiny difference with linux system.

I chmod +x my script, I copied it in /usr/local/bin, I tried to indicate a full path like /Users/david/SCRIPTS/test.sh in the command field.
The better result I've got is a zim pop up windows saying "failed to execute test.sh"  when I copied the script in /usr/local/bin and indicate test.sh in the command field.

**Edit:** 

I tried to use a simple cli command (no shell script), like zenity, but it doesn't wok.

To use zenity or other cli apps, I added /Users/David/Library/Application Support/org.zim-wiki.Zim/zimapp.ini/zimapp.ini
with this ligne :
PATH=<PATH>:/opt/homebrew/bin/

But zim doesn't found the path when I launch the custom tool.

My question is very global : how to run a shell script or cli command on macOS ?

Could you help me ?

Thank you

**Comment by dehesselle** [comment]
I've created a simple demo and it works as expected:

<img width="381" alt="image" src="https://github.com/user-attachments/assets/59d3f4ee-7d59-41a4-9c7f-0441a7191dc6" />

If I click on the "Demo" item, a new empty textfile appears on my desktop. Please try to reproduce that on your end.

Going forward, you need to provide exact details on what you are doing.

Also, modifying `PATH` in `zimapp.ini` is not required to run a fully qualified command. Your current setting is also harmful, you are not following the instructions given in the README. Please undo that before we continue to debug your situation.
